### PR TITLE
Add support for external seed job files.

### PR DESCRIPTION
**Pull request by abayer**
This is obviously quick and dirty, but this seemed like an important
thing to add.

### CODE REVIEW

**Comment by ndeloof** [comment]
LGTM as a "quick and dirty" proof of concept for this purpose
What I had in mind was to introduce some SPI we get configuration content from (https://github.com/jenkinsci/configuration-as-code-plugin/issues/34) which might not be based on filesystem. So CasC would receive a Set of named resource (i.e. a Map name -> stream) and process them. Could then load all `.yaml` files as configuration and consider others a external references. 

Then you can write something like : 
```yaml
jobs:
 +  - ${file:seedJob.groovy}
```

File: src/main/java/org/jenkinsci/plugins/casc/SeedJobConfigurator.java [hunk 1]
```diff
@@ -30,10 +41,26 @@ public String getName() {
...
 
         return null;
     }
+
+    private String getScript(String key) throws IOException {
+        Matcher m = JOB_DSL_FILE_PATTERN.matcher(key);
+        if(m.matches()) {
+            String configPath = System.getProperty(CASC_JENKINS_CONFIG, System.getenv(CASC_JENKINS_CONFIG));
+            String dslFile = m.group(1);
+            if (StringUtils.isBlank(configPath) || Paths.get(dslFile).isAbsolute()) {
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/casc/SeedJobConfigurator.java]: Not sure if it's safe to use the default path. Maybe Jenkins root instead?

**Comment by ewelinawilkosz** [comment]
hej @abayer, can we still count on you? We have Oleg's comment here, and merge conflict - can you take care of that or should we take over?

**Comment by ndeloof** [comment]
dormant

**Pull request closed without merging**